firedrake/functionspace.py

```diff
@@ -284,17 +288,25 @@ def exterior_facet_node_map(self, bcs=None):
                                parent=parent,
                                offset=offset)
 
-    def bottom_nodes(self):
+    def bottom_nodes(self, method='topological'):
         """Return a list of the bottom boundary nodes of the extruded mesh.
         The bottom mask is applied to every bottom layer cell to get the
         dof ids."""
-        return np.unique(self.cell_node_list[:, self.bt_masks[0]])
+        try:
```

Comment:
It is somewhat nitpicking because this is usually quite safe, but I have an aversion against such "clever" use of exceptions, because it might catch problems occurring elsewhere. For example, if `self.bt_masks` has `method`, but `self.bt_masks[method]` has no `0`, then we will still complain about `method`, even though the problem is something else.